File: hw/robalino/u2/Hw 18 Inheritance/mammal.py
from datetime import date
from typing import List


class Mammal(Animal):
    def __init__(self, maximum_hair_size: float, mammary_glands_number: int, id: int, name: str, born_on_date: date, weight: float, sex: str, vaccines: List['Vaccine']):
        super().__init__(id, name, born_on_date, weight, sex, vaccines)
        self._maximum_hair_size = maximum_hair_size
        self._mammary_glands_number = mammary_glands_number

    def breastfeed(self):
        print("Breastfeeding the children")

    @property
    def maximum_hair_size(self) -> float:
        return self._maximum_hair_size

    @maximum_hair_size.setter
    def maximum_hair_size(self, value: float):
        self._maximum_hair_size = value

    @property
    def mammary_glands_number(self) -> int:
        return self._mammary_glands_number

    @mammary_glands_number.setter
    def mammary_glands_number(self, value: int):
        self._mammary_glands_number = value

    def __str__(self) -> str:
        return (f"Mammal({super().__str__()}, maximum_hair_size={self._maximum_hair_size}, "
                f"mammary_glands_number={self._mammary_glands_number})")

File: ws/robalino/u2/Ws15Inheritance/ZooApp/src/main/java/ec/edu/espe/zooapp/model/Mammal.java


package ec.edu.espe.zooapp.model;

import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author Benjamin Robalino <jabasteam>
 */
public class Mammal extends Animal {
    private float maximumHairSize;
    private int mammaryGlandsNumber;
    
    public void breastfeed(){
        System.out.println("Breastfeeding the children ");
    }

    public Mammal(float maximumHairSize, int mammaryGlandsNumber, int id, String name, Date bornOnDate, float weight, char sex, ArrayList<Vaccine> vaccines) {
        super(id, name, bornOnDate, weight, sex, vaccines);
        this.maximumHairSize = maximumHairSize;
        this.mammaryGlandsNumber = mammaryGlandsNumber;
    }

    @Override
    public String toString() {
        return "Mammal{" + super.toString()+ ", maximumHairSize=" + maximumHairSize + ", mammaryGlandsNumber=" + mammaryGlandsNumber + '}';
    }

    /**
     * @return the maximumHairSize
     */
    public float getMaximumHairSize() {
        return maximumHairSize;
    }

    /**
     * @param maximumHairSize the maximumHairSize to set
     */
    public void setMaximumHairSize(float maximumHairSize) {
        this.maximumHairSize = maximumHairSize;
    }

    /**
     * @return the mammaryGlandsNumber
     */
    public int getMammaryGlandsNumber() {
        return mammaryGlandsNumber;
    }

    /**
     * @param mammaryGlandsNumber the mammaryGlandsNumber to set
     */
    public void setMammaryGlandsNumber(int mammaryGlandsNumber) {
        this.mammaryGlandsNumber = mammaryGlandsNumber;
    }
   
}
